kornia/feature/scale_space_detector.py

```diff
@@ -234,3 +236,173 @@ def forward(self, img: Tensor, mask: Optional[Tensor] = None) -> Tuple[Tensor, T
+        model: Module,
+        num_features: int = 2048,
+        config: Detector_config = default_detector_config,
+        ori_module: Optional[Module] = None,
+        aff_module: Optional[Module] = None,
+    ):
+        super().__init__()
+        self.model = model
+        # Load extraction configuration
+        self.num_pyramid_levels = config['pyramid_levels']
+        self.num_upscale_levels = config['up_levels']
+        self.scale_factor_levels = config['scale_factor_levels']
+        self.mr_size = config['s_mult']
+        self.nms_size = config['nms_size']
+        self.nms = NonMaximaSuppression2d((self.nms_size, self.nms_size))
+        self.num_features = num_features
+
+        if ori_module is None:
+            self.ori: Module = PassLAF()
+        else:
+            self.ori = ori_module
+
+        if aff_module is None:
+            self.aff: Module = PassLAF()
+        else:
+            self.aff = aff_module
+
+    def remove_borders(self, score_map, borders: int = 15):
+        """It removes the borders of the image to avoid detections on the corners."""
+        mask = torch.zeros_like(score_map)
+        mask[:, :, borders:-borders, borders:-borders] = 1
+        return mask * score_map
+
+    def detect_features_on_single_level(
+        self, level_img: Tensor, num_kp: int, factor: Tuple[float, float]
+    ) -> Tuple[Tensor, Tensor]:
+        det_map = self.nms(self.remove_borders(self.model(level_img)))
+        device = level_img.device
+        dtype = level_img.dtype
+        yx = det_map.nonzero()[:, 2:].t()
+        scores = det_map[0, 0, yx[0], yx[1]]  # keynet supports only non-batched images
+
+        scores_sorted, indices = torch.sort(scores, descending=True)
+
+        indices = indices[where(scores_sorted > 0.0)]
+        yx = yx[:, indices[:num_kp]].t()
+        current_kp_num = len(yx)
+        xy_projected = yx.view(1, current_kp_num, 2).flip(2) * tensor(factor, device=device, dtype=dtype)
+        scale_factor = 0.5 * (factor[0] + factor[1])
+        scale = scale_factor * self.mr_size * torch.ones(1, current_kp_num, 1, 1, device=device, dtype=dtype)
+        lafs = laf_from_center_scale_ori(xy_projected, scale, zeros(1, current_kp_num, 1, device=device, dtype=dtype))
+        return scores_sorted[:num_kp], lafs
+
+    def detect(self, img: Tensor, mask: Optional[Tensor] = None) -> Tuple[Tensor, Tensor]:
+        # Compute points per level
+        num_features_per_level: List[float] = []
+        tmp = 0.0
+        factor_points = self.scale_factor_levels**2
+        levels = self.num_pyramid_levels + self.num_upscale_levels + 1
+        for idx_level in range(levels):
+            tmp += factor_points ** (-1 * (idx_level - self.num_upscale_levels))
+            nf = self.num_features * factor_points ** (-1 * (idx_level - self.num_upscale_levels))
+            num_features_per_level.append(nf)
+        num_features_per_level = list(map(lambda x: int(x / tmp), num_features_per_level))
```

Comment:
> And as an option for the block, I think this can be
> 
> and the idea here is really to inverse the signal with `-1`?

Not  to inverse the signal, but to split the desired number of keypoints proportionally to the scale level area